kairos-prover/kairos-circuit-logic/src/account_trie.rs

```diff
@@ -110,9 +114,10 @@ impl<S: Store<Account>> AccountTrie<S> {
             )
         })?;
 
-        if sender_account.pubkey != *sender {
-            return Err(("hash collision detected on sender account").into());
-        }
+        // SECURITY ASUMPTION: see Account docs
+        // if sender_account.public_key != *sender {
+        //     return Err(("hash collision detected on sender account").into());
+        // }
```

Comment:
Can we remove this?

Reply:
I just left that anywhere I made the assumption.